clang/lib/CodeGen/CGDeclCXX.cpp

```diff
@@ -333,7 +338,8 @@ void CodeGenFunction::registerGlobalDtorWithLLVM(const VarDecl &VD,
                                                  llvm::FunctionCallee Dtor,
                                                  llvm::Constant *Addr) {
   // Create a function which calls the destructor.
-  llvm::Function *dtorStub = createAtExitStub(VD, Dtor, Addr);
+  llvm::Function *dtorStub =
```

Comment:
So, `createAtExitStub` might return ptrauth constant expression. And the `cast<Function>` here does not fail simply because we do not sign these pointers (yet)?

Reply:
Oh curses, I missed these comments due to earlier insanity this week and/or GitHub mail not be filtered correctly (which I'll check on)

I _think_ these casts occurred a long time ago as merge resolutions or similar.

I meant to ask about what the best solution would be, we _could_ just convert everything to Constant* but I looked at that and it spreads a lot.

It also means losing type system guard for is this being the right type. It feels like the best approach would be to introduce an wrapper type, something akin to 

struct ConstantMaybeSignedFunction {
   FunctionType *FT;
   Constant *Value;
};

That would allow places that do want to reason about the type to continue to do so. In principle this could be MaybeSigned<T>

Reply:
I think it's ok as it is for now. When / if we will really need better abstraction in multiple places we can go via `MaybeSigned<T>` or something like this.